src/runtime/test/hydrate-no-encapsulation.spec.tsx

```diff
@@ -166,6 +166,69 @@ describe('hydrate no encapsulation', () => {
     `);
   });
 
+  it('nested text slot with key', async () => {
+    @Component({ tag: 'cmp-a' })
+    class CmpA {
+      render() {
+        return (
+          <Host key="test">
+            <cmp-b key="my-test-key">light-dom</cmp-b>
+          </Host>
+        );
+      }
+    }
+    @Component({ tag: 'cmp-b' })
+    class CmpB {
+      render() {
+        return (
+          <Host>
+            <slot key="key1"></slot>
+            <footer key="key2"></footer>
+          </Host>
+        );
+      }
+    }
+    // @ts-ignore
+    const serverHydrated = await newSpecPage({
+      components: [CmpA, CmpB],
+      html: `<cmp-a></cmp-a>`,
+      hydrateServerSide: true,
+    });
+    expect(serverHydrated.root).toEqualHtml(`
+      <cmp-a class="hydrated" s-id="1">
+        <!--r.1-->
+        <cmp-b class="hydrated" c-id="1.0.0.0" s-id="2">
+          <!--r.2-->
+          <!--o.1.1-->
+          <!--s.2.0.0.0.-->
+          <!--t.1.1.1.0-->
+          light-dom
+          <footer c-id="2.1.0.1"></footer>
+        </cmp-b>
+      </cmp-a>
+    `);
+
+    // @ts-ignore
```

Comment:
For my own knowledge, why do we need this `@ts-ignore`? If we need it, can we add a justification/reason why we need it in the comment for future us?